roboflow/adapters/rfapi.py

```diff
@@ -158,22 +158,37 @@ def _save_annotation_url(api_key, project_url, name, image_id, job_name, is_pred
     return url
 
 
-def _hosted_upload_url(api_key, project_url, image_path, split):
+def _upload_url(api_key, project_url, **kwargs):
     url = f"{API_URL}/dataset/{project_url}/upload?api_key={api_key}"
-    url += f"&name={os.path.basename(image_path)}&split={split}"
-    url += f"&image={urllib.parse.quote_plus(image_path)}"
+
+    for key, value in kwargs.items():
+        if isinstance(value, list):
+            url += "".join([f"&{str(key)}={str(v)}" for v in value])
+        else:
+            url += f"&{str(key)}={str(value)}"
+
```

Comment:
I think `urllib.parse.urlencode` does exactly it.

```
>>> from urllib.parse import urlencode
>>> urlencode({"a": 1, "b": ["x", "y", "z"]}, doseq=True)
'a=1&b=x&b=y&b=z'
```

```suggestion
    if kwargs:
        querystring = urlencode(kwargs, doseq=True)
        url += f"&{querystring}"
```

Reply:
Yes, it even quotes via `quote_plus`, eliminating the need to explicitly encode image paths using it. Changed at 21a26e21673285dee6884e6df3e8d5251ca0b79f